radiation/python/radlw/taumol_gt4py.py

```diff
@@ -0,0 +1,4957 @@
+    "fac000",
+    "fac100",
+    "fac200",
+    "fac010",
+    "fac110",
+    "fac210",
+]
+
+for var in locvars_int:
+    if var == "ib":
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_2D, DTYPE_INT)
+    else:
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlay, DTYPE_INT)
+
+for var in locvars_flt:
+    if var == "taug":
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlay, type_ngptlw)
+    else:
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlay, DTYPE_FLT)
+
+
+def loadlookupdata(name):
+    """
+    Load lookup table data for the given subroutine
+    """
+    ds = xr.open_dataset("../lookupdata/radlw_" + name + "_data.nc")
+
+    lookupdict = dict()
+    lookupdict_gt4py = dict()
+
+    for var in ds.data_vars.keys():
+        # print(f"{var} = {ds.data_vars[var].shape}")
+        if len(ds.data_vars[var].shape) == 1:
+            lookupdict[var] = np.tile(
+                ds[var].data[None, None, None, :], (npts, 1, nlay, 1)
+            )
+        elif len(ds.data_vars[var].shape) == 2:
+            lookupdict[var] = np.tile(
+                ds[var].data[None, None, None, :, :], (npts, 1, nlay, 1, 1)
+            )
+        elif len(ds.data_vars[var].shape) == 3:
+            lookupdict[var] = np.tile(
+                ds[var].data[None, None, None, :, :, :], (npts, 1, nlay, 1, 1, 1)
+            )
+
+        lookupdict_gt4py[var] = create_storage_from_array(
+            lookupdict[var], backend, shape_nlay, (DTYPE_FLT, ds[var].shape)
+        )
+
+    ds2 = xr.open_dataset("../lookupdata/radlw_ref_data.nc")
+    tmp = np.tile(ds2["chi_mls"].data[None, None, None, :, :], (npts, 1, nlay, 1, 1))
+
+    lookupdict_gt4py["chi_mls"] = create_storage_from_array(
+        tmp, backend, shape_nlay, (DTYPE_FLT, ds2["chi_mls"].shape)
+    )
+
+    return lookupdict_gt4py
+
+
+@stencil(
+    backend=backend,
+    rebuild=rebuild,
+    externals={
+        "nspa": nspa[0],
+        "nspb": nspb[0],
+        "laytrop": indict["laytrop"],
+        "ng01": ng01,
+        "nlay": nlay,
+    },
+)
+def taugb01(
+    pavel: FIELD_FLT,
+    colamt: Field[type_maxgas],
+    colbrd: FIELD_FLT,
+    fac00: FIELD_FLT,
+    fac01: FIELD_FLT,
+    fac10: FIELD_FLT,
+    fac11: FIELD_FLT,
+    jp: FIELD_INT,
+    jt: FIELD_INT,
+    jt1: FIELD_INT,
+    selffac: FIELD_FLT,
+    selffrac: FIELD_FLT,
+    indself: FIELD_INT,
+    forfac: FIELD_FLT,
+    forfrac: FIELD_FLT,
+    indfor: FIELD_INT,
+    minorfrac: FIELD_FLT,
+    scaleminorn2: FIELD_FLT,
+    indminor: FIELD_INT,
+    fracs: Field[type_ngptlw],
+    taug: Field[type_ngptlw],
+    absa: Field[(DTYPE_FLT, (10, 65))],
+    absb: Field[(DTYPE_FLT, (10, 235))],
+    selfref: Field[(DTYPE_FLT, (10, 10))],
+    forref: Field[(DTYPE_FLT, (10, 4))],
+    fracrefa: Field[(DTYPE_FLT, (10,))],
+    fracrefb: Field[(DTYPE_FLT, (10,))],
+    ka_mn2: Field[(DTYPE_FLT, (10, 19))],
+    kb_mn2: Field[(DTYPE_FLT, (10, 19))],
```

Comment:
What do these hard-coded numbers mean. Is 65  nlay+1 or something else? or are they really just arbitrarily sized arrays?

Reply:
They really are just arbitrarily sized arrays. They don't correspond to any of the usual dimension sizes for higher dimensions in the LW scheme. The indices used to read from them are computed in the subroutines